### PR TITLE
Migrate the OpenID module to OpenIddict 5.0

### PR DESCRIPTION
**Pull request by kevinchalet**
OpenIddict 5.0 RTM is expected to ship later this month. This PR reacts to the breaking changes introduced by this new major by adding new store methods are updating the ones whose signature has changed.

(note: while OpenIddict 5.0 supports new features, this PR doesn't expose them via the OC GUI. If there's some interest from the community, it's certainly something we'll want to consider for another PR)

Edit: 5.0 RTM just shipped: https://kevinchalet.com/2023/12/18/openiddict-5-0-general-availability/

### CODE REVIEW

File: src/OrchardCore/OrchardCore.OpenId.Core/YesSql/Models/OpenIdApplication.cs [hunk 1]
```diff
@@ -69,23 +78,26 @@ public class OpenIdApplication
...
-        /// Gets or sets the application type associated with the current application.
+        /// Gets or sets the settings associated with the application.
+        /// </summary>
+        public ImmutableDictionary<string, string> Settings { get; set; }
+            = ImmutableDictionary.Create<string, string>();
+
+        /// <summary>
+        /// Gets or sets the client type associated with the current application.
         /// </summary>
         public string Type { get; set; }
```
Comment (kevinchalet) [inline src/OrchardCore/OrchardCore.OpenId.Core/YesSql/Models/OpenIdApplication.cs:102]: Ideally, we should rename this property to `ClientType` but it's not clear to me what would be the proper way to do it with YesSql /cc @sebastienros

Comment (MichaelPetrinolis) [reply src/OrchardCore/OrchardCore.OpenId.Core/YesSql/Models/OpenIdApplication.cs:102]: And I think there is no need for two sessions/connections, if you get a connection after SaveChangesAsync()

Comment (MikeAlhayek) [reply src/OrchardCore/OrchardCore.OpenId.Core/YesSql/Models/OpenIdApplication.cs:102]: @kevinchalet, you can mark Type as Obsolete and add a new property called ClientType. Setting Type can also set ClientType if ClientType is null.

@MichaelPetrinolis you may want to use `await using` to dispose the connection asynchronously. Also, that could should be call that query using ShellScope.AddDeferredTask to ensure that query is called after everything has been setup (specially during new setup)

Comment (MichaelPetrinolis) [reply src/OrchardCore/OrchardCore.OpenId.Core/YesSql/Models/OpenIdApplication.cs:102]: @MikeAlhayek the Deferred Tasks execute after the migrations complete? If this is the case and the migration fails, subsequently the deferred task will also fail, as the DB won't have the new index table column name (if we also rename the column name 'Type' to 'ClientType' in the index). Maybe adding a DataMigration step that executes after schema migrations (considering the current schema version) would be handy.

Comment (MikeAlhayek) [reply src/OrchardCore/OrchardCore.OpenId.Core/YesSql/Models/OpenIdApplication.cs:102]: Yes differed task will be executed at the end of the request. You can call it after the alter command. If the alter exceptionally failed, the next line will not be called.

Comment (MichaelPetrinolis) [reply src/OrchardCore/OrchardCore.OpenId.Core/YesSql/Models/OpenIdApplication.cs:102]: @MikeAlhayek the issue with the deferred task is that the schema might get updated, but the data are not updated if an error during the update occurs. I think we need a two-step migration process, one to update the schema (existing) and one to run data migrations for the new Schema Version as deferred task.

Comment (kevinchalet) [reply src/OrchardCore/OrchardCore.OpenId.Core/YesSql/Models/OpenIdApplication.cs:102]: @MichaelPetrinolis @MikeAlhayek thanks for your suggestions!

That said, it looks a bit risky and it's likely I won't have too much time to troubleshot any potential issue that may occur due to this change.

If anyone wants to submit a PR to change the name of the `Type` column to `ClientType` (once OpenIddict 5.0 ships next week), please feel free! 😃

**Pull request merged**